common/utils/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -156,6 +156,18 @@ private static void deleteRecursivelyUsingJavaIO(
     }
   }
 
+  /**
+   * Reads basic attributes of a given file, of return null if an I/O error occurs.
+   */
+  private static BasicFileAttributes readFileAttributes(File file) {
+    try {
+      return Files.readAttributes(
```

Comment:
cc @dongjoon-hyun 

1. If the inode itself does not exist, it will not be possible to read the `BasicFileAttributes`.
2. If it is a corrupted symbolic link, the `BasicFileAttributes` can be read, but `file.exists()` will return false.

It seems that scenario 1 was not covered before. I found that the code has logic that attempts to delete a potentially existing target before writing. 

For example: 

https://github.com/apache/spark/blob/91f3fdd25852b43095dd5273358fc394ffd11b66/core/src/main/scala/org/apache/spark/SparkContext.scala#L1879-L1888

Let's test it on macOS first.

Reply:
It's getting a bit late in my time zone, so I need to sleep for 4 hours first. 

Let's take a look at the new test results and decide on the next steps. Of course, if you think the risk is too high, we can also revert this patch first. @dongjoon-hyun  Thanks ~

Reply:
We can wait :)